src/sdl2/log.rs

```diff
@@ -104,3 +121,79 @@ pub fn log(message: &str) {
         crate::sys::SDL_Log(message.as_ptr());
     }
 }
+
+/// Critical log function which takes as priority CRITICAL
+#[doc(alias = "SDL_LogCritical")]
+pub fn log_critical(message: &str, category: Category) {
+    log_with_category(message, category, sdl_log_critical);
+}
+
+/// Debug log function which takes as priority DEBUG
+#[doc(alias = "SDL_LogDebug")]
+pub fn log_debug(message: &str, category: Category) {
+    log_with_category(message, category, sdl_log_debug);
+}
+
+/// Error log function which takes as priority ERROR
+#[doc(alias = "SDL_LogError")]
+pub fn log_error(message: &str, category: Category) {
+    log_with_category(message, category, sdl_log_error);
+}
+
+/// Info log function which takes as priority INFO
+#[doc(alias = "SDL_LogInfo")]
+pub fn log_info(message: &str, category: Category) {
+    log_with_category(message, category, sdl_log_info);
+}
+
+/// Verbose log function which takes as priority VERBOSE
+#[doc(alias = "SDL_LogVerbose")]
+pub fn log_verbose(message: &str, category: Category) {
+    log_with_category(message, category, sdl_log_verbose);
+}
+
+/// Warn log function which takes as priority WARN
+#[doc(alias = "SDL_LogWarn")]
+pub fn log_warn(message: &str, category: Category) {
+    log_with_category(message, category, sdl_log_warn);
+}
+
+/// uses the sdl_log_func to log the message, when category cannot be converted, then Category: Application will be used
+fn log_with_category(message: &str, category: Category, sdl_log_func: unsafe fn(category: libc::c_int, fmt: *const libc::c_char)) {
+    let message = message.replace('%', "%%");
+    let message = CString::new(message).unwrap();
+    let uccategory =  Category::into_ll(category).try_into();
+    let ccategory = match uccategory {
+        Err(_) => Category::into_ll(Category::Application).try_into().unwrap(),
+        Ok(success) => success
+    };
+
+    unsafe {
+        sdl_log_func(ccategory, message.as_ptr());
+    }
+}
+
+unsafe fn sdl_log_critical(category: libc::c_int, fmt: *const libc::c_char){
```

Comment:
There is no point in defining those, juste use `sys::SDL_` functions directly. Users from sdl2 have access to `sdl2::sys` so there is no re-export issue.

Reply:
Ok i have removed those superfluous definitions, with lambda functions inlined in the public functions. 
The lambdas i did to get rid of the variadic parameters of the `sys::SDL_*` versions.
Then i did a cargo fmt to format the codebase and indeed there were some corrections!